core/src/test/java/org/apache/iceberg/hadoop/TestStaticTable.java

```diff
@@ -71,11 +71,20 @@ public void testCannotDoIncrementalScanOnMetadataTable() {
 
     for (MetadataTableType type : MetadataTableType.values()) {
       Table staticTable = getStaticTable(type);
-      AssertHelpers.assertThrows(
-          "Static tables do not support incremental scans",
-          UnsupportedOperationException.class,
-          String.format("Cannot incrementally scan table of type %s", type),
-          () -> staticTable.newScan().appendsAfter(1));
+
+      if (type.equals(MetadataTableType.POSITION_DELETES)) {
+        AssertHelpers.assertThrows(
+            "POSITION_DELETES table does not support TableScan",
```

Comment:
So I think with the new interface adoption, PositionDeleteTable is implementing newBatchScan() instead of newScan() , and throwing exception for the latter.  (As TableScan is bound with FileScanTask)